akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -239,6 +239,63 @@ private[akka] final case class Fold[In, Out](zero: Out, f: (Out, In) ⇒ Out, de
 /**
  * INTERNAL API
  */
+private[akka] final case class Intersperse[T](start: Option[T], inject: T, end: Option[T]) extends StatefulStage[T, T] {
```

Comment:
I can make it a PushPullStage I guess - would that be much more efficient?
Is `emit` available on those?